backend/handlers/get_edge_info.go

```diff
@@ -34,55 +70,102 @@ func getRelationshipReason(tx neo4j.Transaction, edgeID int) (neo4j.Result, erro
 	return record, nil
 }
 
-func AssumeRoleQuery(tx neo4j.Transaction, edgeID int) (neo4j.Result, error) {
+func IdentityPolicyQuery(tx neo4j.Transaction, edgeID int) (neo4j.Result, error) {
 	params := map[string]interface{}{
 		"edge_id": edgeID,
 	}
-	record, err := tx.Run(
-		`match ()-[r]->() where ID(r)=$edge_id return type(r) as type;`,
+	records, err := tx.Run(
+		`match (n)-[r]->(t) where ID(r)=$edge_id 
+		WITH n
+		MATCH (n)-[:POLICY]-(p:AWSPolicy)-[:STATEMENT]-(s:AWSPolicyStatement)
+		WITH p, collect({resource:s.resource,
+			notaction:s.notaction,
+			action:s.action,
+			effect:s.effect,
+			condition:s.condition,
+			notresource:s.notresource,
+			sid:s.sid
+			}) as statement_lst
+		return {arn: p.id,type: p.type} as policy, statement_lst`,
 		params)
 
 	if err != nil {
 		return nil, err
 	}
 
-	return record, nil
+	return records, nil
 }
 
-func getEdgeType(tx neo4j.Transaction, edgeID int) (string, error) {
+func getEdgeParameters(tx neo4j.Transaction, edgeID int) (EdgeParameters, error) {
 	params := map[string]interface{}{
 		"edge_id": edgeID,
 	}
 	records, err := tx.Run(
-		`match ()-[r]->() where ID(r)=$edge_id return type(r) as type;`,
+		`match (s)-[r]->(t) where ID(r)=$edge_id 
+		return type(r) as type, labels(s) as srcLabels, labels(t) as targetLabels;`,
 		params)
 
 	if err != nil {
-		return "None", err
+		return EdgeParameters{}, err
 	}
 
-	var resultEdgeType string
+	var edgeParameters EdgeParameters
 
 	for records.Next() {
 		record := records.Record()
+
 		edgeType, _ := record.Get("type")
+
 		edgeTypeStr, ok := edgeType.(string)
 		if !ok {
-			return "None", fmt.Errorf("edge type %v must be a string", edgeType)
+			return EdgeParameters{}, fmt.Errorf("edge type %v must be a string", edgeType)
+		}
+		srcLabels, _ := record.Get("srcLabels")
+		srcLabelLst, err := castLabelLst(srcLabels)
+
+		if err != nil {
+			return EdgeParameters{}, err
+		}
+
+		targetLabels, _ := record.Get("targetLabels")
+
+		targetLabelLst, err := castLabelLst(targetLabels)
+		if err != nil {
+			return EdgeParameters{}, err
+		}
+
+		edgeParameters = EdgeParameters{
+			edgeType:     edgeTypeStr,
+			srcLabels:    srcLabelLst,
+			targetLabels: targetLabelLst,
 		}
-		resultEdgeType = edgeTypeStr
+
 	}
-	return resultEdgeType, nil
+	return edgeParameters, nil
 }
 
-func getPrivelegeEscalationReason(tx neo4j.Transaction, edgeID int) (interface{}, error) {
+func castLabelLst(labels interface{}) ([]string, error) {
+
+	var resutLabelLst []string
+	labelLst, ok := labels.([]interface{})
+	if !ok {
+		return nil, fmt.Errorf("labels must be a slice of strings")
+	}
+	for _, label := range labelLst {
+		labelStr, _ := label.(string)
```

Comment:
check this is valid and return error if not?

Reply:
done